classes/quiz_settings.php

```diff
@@ -262,105 +262,93 @@ private function compute_config() {
         }
 
         // Export and save the config, ready for DB.
         $this->set('config', $this->plist->to_xml());
     }
 
     /**
-     * Creates an initial plist, then applies all settings associated with the quiz.
+     * Case for USE_SEB_CONFIG_MANUALLY. This creates a plist and applies all settings from the posted form, along with
+     * some defaults.
      */
-    private function process_default_case() {
+    private function process_seb_config_manually() {
         // If at any point a configuration file has been uploaded and parsed, clear the settings.
         $this->plist = new property_list();
-        $this->progress_default_standard_settings();
+
+        $this->process_bool_settings();
+        $this->process_quit_password_settings();
+        $this->process_quit_url_from_settings();
+        $this->process_url_filters();
+        $this->process_required_enforced_settings();
+
+        // One of the requirements for USE_SEB_CONFIG_MANUALLY is setting examSessionClearCookiesOnStart to false.
+        $this->plist->set_or_update_value('examSessionClearCookiesOnStart', new CFBoolean(false));
     }
 
     /**
-     * Creates an initial plist from the template uploaded, then applies all settings associated with the quiz.
+     * Case for USE_SEB_TEMPLATE. This creates a plist from the template uploaded, then applies the quit password
+     * setting and some defaults.
      */
-    private function process_template_settings() {
+    private function process_seb_template() {
         $template = template::get_record(['id' => $this->get('templateid')]);
         $this->plist = new property_list($template->get('content'));
 
-        // Process quit settings.
-        $this->process_quit_settings();
-
-        // Add the sensible default options to the configuration and exported SEB files.
+        $this->process_quit_password_settings();
+        $this->process_quit_url_from_template_or_config();
         $this->process_required_enforced_settings();
     }
 
     /**
-     * This function processes the elements that have been set via the modform fields.
-     *
-     * It's called during case USE_SEB_TEMPLATE and USE_SEB_CONFIG_MANUALLY.
+     * Case for USE_SEB_UPLOAD_CONFIG. This creates a plist from an uploaded configuration file, then applies the quiz
+     * password settings and some defaults.
      */
-    private function progress_default_standard_settings() {
-        // Process all settings that are boolean.
-        $this->process_bool_settings();
-
-        // Process quit settings.
-        $this->process_quit_settings();
+    private function process_seb_upload_config() {
+        $file = settings_provider::get_module_context_sebconfig_file($this->get('cmid'));
 
-        // Add all the URL filters.
-        $this->process_url_filters();
+        // If there was no file, create an empty plist so the rest of this wont explode.
+        if (empty($file)) {
+            $this->plist = new property_list();
```

Comment:
Maybe it's better to explode? Otherwise it will be broken and all users will get broken seb config. I feel like exception here should be ok. Ideally this never should happen, but if it has  happened that means that something horribly went wrong and we want to let the user know and terminate saving.